src/brokercollect/broker_collection.go

```diff
@@ -86,66 +86,75 @@ func brokerWorker(brokerChan <-chan int, collectedTopics []string, wg *sync.Wait
 		}
 
 		// Create Broker
-		b, err := createBroker(brokerID, zkConn, i)
+		brokers, err := createBrokers(brokerID, zkConn, i)
 		if err != nil {
 			continue
 		}
 
-		// Populate inventory for broker
-		if args.GlobalArgs.All() || args.GlobalArgs.Inventory {
-			log.Debug("Collecting inventory for broker %q", b.Entity.Metadata.Name)
-			if err := populateBrokerInventory(b); err != nil {
-				continue
+		for _, broker := range brokers {
+			// Populate inventory for broker
+			if args.GlobalArgs.All() || args.GlobalArgs.Inventory {
+				log.Debug("Collecting inventory for broker %s", broker.Entity.Metadata.Name)
+				if err := populateBrokerInventory(broker); err != nil {
+					continue
+				}
+				log.Debug("Done Collecting inventory for broker %s", broker.Entity.Metadata.Name)
 			}
-			log.Debug("Done collecting inventory for broker %q", b.Entity.Metadata.Name)
-		}
 
-		// Populate metrics for broker
-		if args.GlobalArgs.All() || args.GlobalArgs.Metrics {
-			log.Debug("Collecting metrics for broker %q", b.Entity.Metadata.Name)
-			if err := collectBrokerMetrics(b, collectedTopics); err != nil {
-				continue
+			// Populate metrics for broker
+			if args.GlobalArgs.All() || args.GlobalArgs.Metrics {
+				log.Debug("Collecting metrics for broker %s", broker.Entity.Metadata.Name)
+				if err := collectBrokerMetrics(broker, collectedTopics); err != nil {
+					continue
+				}
+				log.Debug("Done Collecting metrics for broker %s", broker.Entity.Metadata.Name)
 			}
-			log.Debug("Done Collecting metrics for broker %q", b.Entity.Metadata.Name)
 		}
 	}
 }
 
-// Creates and populates a broker struct with all the information needed to
+// Creates and populates an array of broker structs with all the information needed to
 // populate inventory and metrics.
-func createBroker(brokerID int, zkConn zookeeper.Connection, i *integration.Integration) (*broker, error) {
+func createBrokers(brokerID int, zkConn zookeeper.Connection, i *integration.Integration) ([]*broker, error) {
```

Comment:
I'd probably rename this from `createBrokers` to something like `createBrokerVariants` or `createBrokerConnectionVariants` to be clear that all the brokers returned by this function are the same broker, but with different connection params

Reply:
Additionally, with an edit the doc comment to make this clear